firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift

```diff
@@ -190,20 +189,36 @@ class TrackingProtectionViewController: UIViewController,
     // MARK: Redux
     func newState(state: TrackingProtectionState) {
         trackingProtectionState = state
-        if trackingProtectionState.showTrackingProtectionSettings {
-            enhancedTrackingProtectionMenuDelegate?.settingsOpenPage(settings: .contentBlocker)
-        } else if trackingProtectionState.showDetails {
-            showTrackersDetailsController()
-        } else if trackingProtectionState.showBlockedTrackers {
-            showBlockedTrackersController()
-        } else if trackingProtectionState.showsClearCookiesAlert {
-            onTapClearCookiesAndSiteData()
-        } else if trackingProtectionState.shouldClearCookies {
+        if let navigateTo = state.navigateTo {
+            switch navigateTo {
+            case .home:
+                navigationController?.popToRootViewController(animated: true)
+            case .back:
+                navigationController?.popViewController(animated: true)
+            case .close:
+                enhancedTrackingProtectionMenuDelegate?.didFinish()
+            case .settings:
+                enhancedTrackingProtectionMenuDelegate?.settingsOpenPage(settings: .contentBlocker)
+            }
+        }
```

Comment:
for these cases can we utilize the BVCState, since it handles the high level navigation for this view? This avoids having repeated code + navigation in various screens. For example, there may be multiple views that navigate to settings, so we can have the BVCState handle that as its the parent view to close ETP and open settings. 

We can look at adding some more cases to `NavigationDestination` to help handle the navigation to the respective places. What do you think?

Here's an example of proposing how to do general navigation on a higher level since there were some similarities between menu + home navigation that can be handled by browserVC. 

Reference: https://github.com/mozilla-mobile/firefox-ios/pull/22597/files